File: src/main/java/com/epam/mentoring/kotlin/DogBreedApiClient.java
package com.epam.mentoring.kotlin;

import java.util.List;
import java.util.Map;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

@Component
public class DogBreedApiClient {
    private static final String DOG_BREED_API_URL = "https://dog.ceo/api/breeds/list/all";
    private final RestTemplate restTemplate;

    public DogBreedApiClient(final RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }

    public Map<String, List<String>> getBreeds() throws Exception {
        ResponseEntity<DogBreedApiResponse> response = restTemplate.getForEntity(DOG_BREED_API_URL, DogBreedApiResponse.class);
        if(200 != response.getStatusCode().value()) {
            throw new Exception();
        }
        return response.getBody().message();
    }
}

File: src/main/kotlin/com/epam/mentoring/kotlin/DogBreedApiClient.kt
package com.epam.mentoring.kotlin

import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.awaitBody

@Component
class DogBreedApiClient(
    private val webClient: WebClient
) {
    companion object {
        private const val BASE_URL = "https://dog.ceo/api"
    }

    suspend fun getAllBreeds(): DogBreedApiResponse {
        return webClient.get()
            .uri("$BASE_URL/breeds/list/all")
            .retrieve()
            .awaitBody()
    }

    suspend fun getBreedImage(breed: String): DogBreedImageResponse {
        return webClient.get()
            .uri("$BASE_URL/breed/$breed/images/random")
            .retrieve()
            .awaitBody()
    }
} 